s3/src/main/java/com/instaclustr/kafka/connect/s3/TransferManagerProvider.java

```diff
@@ -30,9 +36,34 @@ public static AmazonS3ClientBuilder getS3ClientBuilderWithRegionAndCredentials(f
         String accessKey = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_ACCESS_KEY_ID);
         String secret = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_SECRET_KEY);
         String region = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_REGION);
+        String roleArn = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_IAM_ROLE_ARN);
 
-        AmazonS3ClientBuilder clientBuilder = AmazonS3ClientBuilder.standard()
-                .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secret)));
+        AmazonS3ClientBuilder clientBuilder;
+
+        if (roleArn == null || StringUtils.isBlank(roleArn)) {
+            // authenticate with access key/secret
+            AWSStaticCredentialsProvider awsStaticCredentialsProvider = new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secret));
+            clientBuilder = AmazonS3ClientBuilder.standard()
+                    .withCredentials(awsStaticCredentialsProvider);
+        } else {
+            // authenticate with access key/secret, then assume role
+            AWSSecurityTokenService awsSecurityTokenService = AWSSecurityTokenServiceClientBuilder.standard()
+                    .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secret)))
```

Comment:
new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secret)) could be built before entering the if

Reply:
similarly for the client creation

Reply:
done